junit-jupiter-api/src/main/java/org/junit/jupiter/api/DisplayNameGenerator.java

```diff
@@ -82,14 +86,41 @@ public interface DisplayNameGenerator {
 	String generateDisplayNameForClass(Class<?> testClass);
 
 	/**
-	 * Generate a display name for the given {@link Nested @Nested} inner test class.
+	 * Generate a display name for the given {@link Nested @Nested} inner test
+	 * class.
 	 *
-	 * <p>If it returns {@code null}, the default display name generator will be used instead.
+	 * <p>If it returns {@code null}, the default display name generator will be
+	 * used instead.
+	 *
+	 * @param nestedClass the class to generate a name for; never {@code null}
+	 * @return the display name for the nested class; never blank
+	 * @deprecated in favor of {@link #generateDisplayNameForNestedClass(List, Class)}
+	 */
+	@SuppressWarnings("DeprecatedIsStillUsed")
+	@API(status = DEPRECATED, since = "5.12")
+	@Deprecated
+	default String generateDisplayNameForNestedClass(@SuppressWarnings("unused") Class<?> nestedClass) {
+		throw new UnsupportedOperationException(
+			"Implement generateDisplayNameForNestedClass(List<Class<?>>, Class<?>) instead");
+	}
+
+	/**
+	 * Generate a display name for the given {@link Nested @Nested} inner test
+	 * class.
+	 *
+	 * <p>If it returns {@code null}, the default display name generator will be
+	 * used instead.
 	 *
+	 * @param enclosingInstanceTypes the runtime types of the enclosing
```

Comment:
We need to document the order of the enclosing instance types list (top-down, bottom-up, whether the list includes/excludes the _current_ nested class, etc.), either in every affected method or at the class level (and then reference the class-level documentation).

Based on the changes made to `IndicativeSentences`, it appears that the order is top-down.

Reply:
Done in 100c3858458b47fe58d0b54aad01ef9971b2c57f